src/org/opensolaris/opengrok/configuration/Configuration.java

```diff
@@ -83,7 +83,17 @@
     private boolean historyCacheInDB;
 
     private int messageLimit;
+    /**
+     * Directory with authorization plugins. Default value is
+     * dataRoot/../plugins (can be /var/opengrok/plugins if dataRoot is
+     * /var/opengrok/data).
+     */
     private String pluginDirectory;
+    /**
+     * Enable watching the plugin directory for changes in real time. Suitable
+     * for development.
+     */
+    private boolean authorizationWatchdog;
```

Comment:
> given this will be present in configuration.xml, the name of the var should signify the type, what about authorizationWatchdogEnabled or authorizationWatchdogOn or similar ?

changed

> Also, I'd expect some mention (ideally control of) the refresh/watch period.

There is no way to control this. It is done automatically with the WatchService and that is implementation dependent how the service watch for changes.

In case you meant this:

```java
Thread.sleep(THREAD_SLEEP_TIME); // experimental wait if file is being written right now
AuthorizationFramework.getInstance().reload();
```

It means something different than the refresh period. It is set to 2s and I think that it is not necessary to control this and also that is is enough.

I added a comment to this method (which has already been there for the `authorizationWatchdogEnabled` variable in configuration).